src/odemis/acq/stitching/test/registrar_test.py

```diff
@@ -0,0 +1,295 @@
+        img = Image.open("images/test.png")
+        cropped1 = img.crop((0,0,400,400))
+        cropped2 = img.crop((322,4,722,404))
+  
+        registrar = ShiftRegistrar()
+        tile1 = model.DataArray(np.array(cropped1.convert("L")),{
+                model.MD_PIXEL_SIZE: [1/20,1/20],  # m/px
+                model.MD_POS: (200/20, img.size[1]/20-200/20),  # m
+            })
+        tile2 = model.DataArray(np.array(cropped2.convert("L")),{
+                model.MD_PIXEL_SIZE: [1/20,1/20],  # m/px
+                model.MD_POS: (520/20, img.size[1]/20-200/20),  # m
+            })
+        registrar.addTile(tile1)
+        registrar.addTile(tile2)
+        calculatedPositions = registrar.getPositions()[0]
+        self.assertAlmostEqual(calculatedPositions[1][0],522/20,places=1)
+        self.assertAlmostEqual(calculatedPositions[1][1],img.size[1]/20-204/20,places=1)          
+        
+def decomposeImage(img, overlap=0.1, numTiles = 5, method="horizontalLines"):
+    """ 
+    Decomposes image into tiles for testing. The tiles overlap and their center positions are subject to random noise.
+    Returns list of tiles and list of the actual positions. 
+    img: square image obtained by PIL module
+    numTiles: number of desired tiles in each direction
+    method: acquisition method, "horizontalLines" scans image by row and starts at the left for each row,
+    "verticalLines" scans image by columns starting at the top for each row, and "horizontalZigzag" scans 
+    a row, then scans the next row in reverse, etc. mimicking the behaviour of DELMIC microscopes. 
+    """
+    
+    tileSize = int(img.size[0]/numTiles)
+    
+    pos = []
+    tiles = []
+    for i in range(numTiles):
+        for j in range(numTiles):
+            # Positions top left            
+            if method == "verticalLines":
+                posX = int(i*(1-overlap)*tileSize)
+                posY = int(j*(1-overlap)*tileSize)
+            elif method == "horizontalLines":
+                posX = int(j*(1-overlap)*tileSize)
+                posY = int(i*(1-overlap)*tileSize)
+            elif method == "horizontalZigzag":  
+                if i%2 == 0:
+                    posX = int(j*(1-overlap)*tileSize)
+                else:
+                    posX = int((numTiles-j-1)*(1-overlap)*tileSize) # reverse direction for every second row
+                posY = int(i*(1-overlap)*tileSize)
+            
+            yMax = numTiles*(1-overlap)*tileSize    
+            md = {
+                model.MD_PIXEL_SIZE: [0.01,0.01],  # m/px
+                model.MD_POS: ((posX+tileSize/2)*0.01,(yMax-posY+tileSize/2)*0.01),  # m
+            }
+             
+            # Add noise
+            maxNoise = int(0.2*overlap*tileSize)
+            random.seed(1)
+            noise = [random.randrange(-maxNoise,maxNoise) for _ in range(2)]
+            if i>0 or j>0:
+                posX = max(0,int(posX + noise[0]))
```

Comment:
Please add a comment about why there is no need to check that the position is too big. (because the original position of the tile includes the overlap, so there is always at least 1* overlap of margin)